typeclasses/chargen.py

```diff
@@ -0,0 +1,587 @@
+                        "goto": "menunode_allocate_mana"})
+        return (text, help), options
+    else:
+        if tr.MAG.actual > 0:
+            output = "Final Mana Values:\n"
+            output += "  |wWhite Mana|n: |w{}|n\n".format(tr.WM.actual)
+            output += "  |xBlack Mana|n: |w{}|n\n\n".format(tr.BM.actual)
+        else:
+            output = ""
+        # TODO: implement spells; add level 0 spell cmdsets here
+
+        archetypes.calculate_secondary_traits(char.traits)
+        archetypes.finalize_traits(char.traits)
+        skills.apply_skills(char)
+        return menunode_allocate_skills(caller, output)
+
+
+def menunode_allocate_skills(caller, raw_input):
+    """Skill -1 counter allocation menu node."""
+    char = caller.new_char
+    sk = char.skills
+    total = 3
+    counts = {1: 'one', 2: 'two', 3: 'three'}
+
+    plusses = (ceil(char.traits.INT.actual / 3.0) -
+               sum(sk[s].plus for s in skills.ALL_SKILLS))
+    minuses = total - sum(sk[s].minus for s in skills.ALL_SKILLS)
+
+    text = ""
+    if raw_input.isdigit() and int(raw_input) <= len(skills.ALL_SKILLS):
+        skill = sk[skills.ALL_SKILLS[int(raw_input)-1]]
+        if minuses:
+            if skill.actual - skill.minus - 1 > 0:
+                skill.minus += 1
+                minuses -= 1
+            else:
+                text += "|rSkills cannot be reduced below one.|n\n"
+        elif plusses:
+            if skill.actual + skill.plus + 1 <= 10:
+                skill.plus += 1
+                plusses -= 1
+            else:
+                text += "|rSkills cannot be increased above ten.|n\n"
+
+    if plusses or minuses:
+        text += raw_input if raw_input and raw_input[0] == 'F' else ""
+        text += "Your ability to perform actions in Ainneve is\n"
+        text += "tied to your character's skills. Your current skills:\n"
+
+        data = []
+        for i in xrange(3):
+            data.append([_format_skill_3col(sk[s])
+                         for s in skills.ALL_SKILLS[i::3]])
+        table = EvTable(header=False, table=data)
+
+        text += "{}\n".format(table)
+        if minuses:
+            text += "Please allocate |w{}|n |m'-1'|n counter{}.".format(
+                        counts[minuses],
+                        's' if minuses != 1 else '')
+        elif plusses:
+            text += "Please allocate |w{}|n |m'+1'|n counter{}.".format(
+                        counts[plusses],
+                        's' if plusses != 1 else '')
+
+        help = "Skill allocation is a two step process. First, you\n"
+        help += "distribute three '-1' counters across your skills,\n"
+        help += "then a number of '+1' counters equal to your Intelligence\n"
+        help += "divided by 3."
+
+        options = [{"desc": sk[s].name,
+                    "goto": "menunode_allocate_skills"}
+                   for s in skills.ALL_SKILLS]
+
+        def clear_skills(c, i):
```

Comment:
I looked at all of the callbacks throughout the menu, and nearly all of them accept only one argument, `s`, because throughout this menu, `caller` is the session. 

I like that better for callbacks, as they rarely use the argument anyway, and could easily be modified to never use it.